hooks/src/index.js

```diff
@@ -181,15 +181,43 @@ export function useReducer(reducer, initialState, init) {
 
 		hookState._component = currentComponent;
 
-		hookState._component.shouldComponentUpdate = () => {
-			if (!hookState._nextValue) return true;
+		if (!hookState._component._hasScuFromHooks) {
+			hookState._component.__hooks._hasScuFromHooks = true;
+			const prevScu = hookState._component.shouldComponentUpdate;
+			hookState._component.shouldComponentUpdate = (p, s, c) => {
+				const stateHooks = hookState._component.__hooks._list.filter(
+					x => x._component
+				);
+				const allHooksEmpty = stateHooks.every(x => !x._nextValue);
```

Comment:
For state updates the hook value could be `0` or `""` which would be interpreted as empty here. Let's check for `undefined` instead here to avoid casting errors.

```suggestion
				const allHooksEmpty = stateHooks.every(x => x._nextValue === undefined);
```

Reply:
_nextValue in this case is an array like the normal state hook containing [value, updater] so using ! is the least bytes and correct as it's array or undefined

Reply:
Makes sense, I was confusing it with the actual value. Thanks for correcting me 👍